File: mvc-architecture/mvc-arch/src/main/java/org/example/mvc/controller/UserController.java
package org.example.mvc.controller;

import org.example.mvc.annotation.Controller;
import org.example.mvc.annotation.Inject;
import org.example.mvc.service.UserService;

@Controller
public class UserController {
    private final UserService userService;

    @Inject
    public UserController(UserService userService) {
        this.userService = userService;
    }

    public UserService getUserService() {
        return userService;
    }
}

File: Week3/src/main/kotlin/com/example/gameduo/controller/UserController.kt
package com.example.gameduo.controller

import com.example.gameduo.dto.UserDTO
import com.example.gameduo.service.UserService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RestController


@RestController
class UserController(private val userService: UserService) {
    @PostMapping("/user")
    internal fun createUser(): UserDTO =
        userService.createUser()

    @GetMapping("/user/{userId}")
    internal fun getUser(@PathVariable("userId") userId: Long): UserDTO =
        userService.getUser(userId)
}